### PR TITLE
Add basic analyics:

### LINKED ISSUE #69: Add Analytics
Add Analytics to storefront. 

Blocked by Impressum + Privacy

### PR DESCRIPTION
**Pull request by Lebski**
*** Integration blocked by cookie policy and privacy statement *** 

Set up Google Analytics in develop and production Mode
- Track site navigation
- Track search
- Track social media links

### CODE REVIEW

File: services/analytics.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import ReactGA from "react-ga";
```
Comment (connorlanigan) [inline services/analytics.tsx:1]: `react-ga` is missing in the package.json file

File: pages/_app.tsx [hunk 2]
```diff
@@ -73,6 +74,8 @@ const SmallBusinessHeroApp = ({Component, pageProps}) => {
 
 	const classes = useStyles();
 
+	initializeAnalytics();
```
Comment (connorlanigan) [inline pages/_app.tsx]: This code runs each time the component is rendered. Currently, the root component is not likely to rerender, but we will not know when that changes. You can move this line outside the `SmallBusinessHeroApp` component to make sure it executes only once.

File: services/analytics.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  if (typeof window !== 'undefined') {
+    // On local, setup testMode, so data won't go to analytics dashboard
+    if (process.env.NODE_ENV === "development") {
+      ReactGA.initialize('UA-162524264-1', { testMode: true, debug: true });
+    } else {
+      ReactGA.initialize('UA-162524264-1');
+    }
+
+  ReactGA.pageview(window.location.pathname + window.location.search);
+    }
```
Comment (connorlanigan) [inline services/analytics.tsx]: improperly indented closing bracket

File: services/analytics.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+// Send event if user uses search bar and log place id 
+export function analyticsSearchLocation(placeID: string) {
+  ReactGA.event({
+      category: 'user-input',
+      action: 'search-location',
+      label: placeID,
+    });
```
Comment (connorlanigan) [inline services/analytics.tsx]: improperly indented closing bracket

File: services/analytics.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+	// Send event event if social-media link is clicked
+export function analyticsSocialMediaRef(eventType: string, storeId: string) {
+  ReactGA.event({
+      category: 'social-media-ref',
+      action: eventType,
+      label: storeId,
+    });
```
Comment (connorlanigan) [inline services/analytics.tsx]: improperly indented closing bracket

File: pages/stores/[storeId].tsx [hunk 6]
```diff
@@ -16,6 +16,7 @@ import {faGlobe} from '@fortawesome/free-solid-svg-icons';
 import {faEnvelope} from '@fortawesome/free-regular-svg-icons';
 import {Link} from '@material-ui/core';
 import useStyles from '../../assets/styles/stores_styles';
+import {analyticsSocialMediaRef} from '../../services/analytics';
```
Comment (connorlanigan) [inline pages/stores/[storeId].tsx]: minor: this is formatted differently than in the other files (missing spaces around the import name). You can use the popular [prettier](https://prettier.io/) code formatter to auto-format your code uniformly.

**Review by connorlanigan — CHANGES_REQUESTED** [review]

**Comment by connorlanigan** [comment]
Hi @Lebski , please don't resolve code review conversations without adressing them in some way (code change or comment).

**Comment by Lebski** [comment]
@connorlanigan Sorry, I thought you have already resolved them.

File: services/analytics.tsx (hunk 1 above)
Comment (Lebski) [reply services/analytics.tsx:1]: Resolved in e9332a6

File: pages/_app.tsx (hunk 2 above)
Comment (Lebski) [reply pages/_app.tsx]: Resolved in  0e01fea

File: services/analytics.tsx (hunk 3 above)
Comment (Lebski) [reply services/analytics.tsx]: Resolved in 324a621

File: services/analytics.tsx (hunk 4 above)
Comment (Lebski) [reply services/analytics.tsx]: Resolved in 324a621

File: services/analytics.tsx (hunk 5 above)
Comment (Lebski) [reply services/analytics.tsx]: Resolved in 324a621

File: pages/stores/[storeId].tsx (hunk 6 above)
Comment (Lebski) [reply pages/stores/[storeId].tsx]: Resolved in 324a621

**Review by connorlanigan — APPROVED** [review]

**Pull request merged**